Testing this against eval debugger, with:
* Original test (build macro + build macro + Context.parseInlineString)
* Expr macro variant (expr macro + build macro + Context.parseInlineString)

Note that for these, Haxe compiler doesn't show the full stack:
```
[ERROR] src/Bar.hx:1: characters 1-2

 1 | @:build(Macro.build())
   | ^
   | Unexpected &
```

Eval debugger shows the whole stack, like current test (build macro + build macro + `throw`):
![image](https://github.com/HaxeFoundation/haxe/assets/6101998/b187cd7e-0c21-47b2-91ed-1f30d6642361)

I updated `kind_name` (locally) to something I think is a bit better:
![image](https://github.com/HaxeFoundation/haxe/assets/6101998/bbf15a48-f71c-4f27-b922-877c62937d5e)

And also works for expression macro => build macro:
![image](https://github.com/HaxeFoundation/haxe/assets/6101998/c0296e6c-cdfe-4758-83aa-7b8ae3bb4e18)

---

Haxe compiler not showing the stack for `Context.parseInlineString` example now seems weird...

--- a/src/macro/eval/evalContext.ml
+++ b/src/macro/eval/evalContext.ml
@@ -334,7 +334,7 @@ let kind_name eval kind =
 		| EKMethod(i1,i2) ->
 			Printf.sprintf "%s.%s" (rev_hash i1) (rev_hash i2)
 		| EKMacro(i1,i2) ->
-			Printf.sprintf "%s:%s" (rev_hash i1) (rev_hash i2)
+			Printf.sprintf "Macro call: %s.%s" (rev_hash i1) (rev_hash i2)
 		| EKLocalFunction i ->
 			begin match env with
 			| None -> Printf.sprintf "localFunction%i" i

--- a/src/macro/eval/evalExceptions.ml
+++ b/src/macro/eval/evalExceptions.ml
@@ -122,6 +122,17 @@ let catch_exceptions ctx ?(final=(fun() -> ())) f p =
 		eval.caught_exception <- vnull;
 		Option.may (build_exception_stack ctx) env;
 		eval.env <- env;
+
+		(* Careful: We have to get the message before resetting the context because toString() might access it. *)
+		let get_stack ctx =
+			let stack = match eval_stack with
+				| [] -> []
+				| l when p' = null_pos -> l (* If the exception position is null_pos, we're "probably" in a built-in function. *)
+				| _ :: l -> l (* Otherwise, ignore topmost frame position. *)
+			in
+			get_exc_error_stack ctx stack
+		in
+
 		if is v key_haxe_macro_Error then begin
 			let v1 = field v key_exception_message in
 			let v2 = field v key_pos in
@@ -139,7 +150,9 @@ let catch_exceptions ctx ?(final=(fun() -> ())) f p =
 							end else
 								Error.raise_typing_error (Printf.sprintf "Unexpected value where haxe.macro.Error was expected: %s" (s_value 0 v).sstring) null_pos
 						) (EvalArray.to_list sub)
-				| _ -> []
+				| _ ->
+					let stack = get_stack ctx in
+					List.map (fun p -> (Error.Custom "Called from here", p)) (List.rev stack)
 			in
 			reset_ctx();
 			final();
@@ -168,13 +181,7 @@ let catch_exceptions ctx ?(final=(fun() -> ())) f p =
 				| v ->
 					Error.raise_typing_error (Printf.sprintf "Invalid exception value where string was expected: %s" (s_value 0 v).sstring) null_pos
 		end else begin
-			(* Careful: We have to get the message before resetting the context because toString() might access it. *)
-			let stack = match eval_stack with
-				| [] -> []
-				| l when p' = null_pos -> l (* If the exception position is null_pos, we're "probably" in a built-in function. *)
-				| _ :: l -> l (* Otherwise, ignore topmost frame position. *)
-			in
-			let stack = get_exc_error_stack ctx stack in
+			let stack = get_stack ctx in
 			reset_ctx();
 			final();
 			let p = if p' = null_pos then p else p' in

